### ISSUE TITLE
getBucked data inaccurate.

### ISSUE THREAD

**Issue by cryptoinminutes** [BitMEX/api-connectors#144]
Im trying to get the last 20,  5-minutes candles from Bitmex but when i enter this same request through the Bitmex Explorer (https://www.bitmex.com/api/explorer/) or compare the results to the chart they differ. 
Is there a reason for this?

```
.client.Trade.Trade_getBucketed(
            binSize='5m',
            partial=True,
            symbol='XBT',
            count=20,  
            reverse=True
        ).result()[0]
```

**Comment by 11bblandin** [comment]
The results are as if 'reverse' is set to True, regardless of what is input. Is there a workaround for this?

**Comment by ryanfox** [comment]
If you specify __any__ value for `reverse` (even `False`), the resulting request will have `reverse=True`. As a workaround, try removing that parameter:

    client.Trade.Trade_getBucketed(binSize='5m', partial=True, symbol='XBT', count=20).result()[0]

**Comment by h0wXD** [comment]
You are correct, I am facing the same issue on LIVE. The getBucketed data differs from the one showing on the chart.
For example for 2018-09-05T16:36:00.000Z XBTUSD
bucketed: `O: 6978.5 H: 6979 L: 6977.5 C: 6977.5`
chart: `O: 6977.5 H: 6983 L: 6977.5 C: 6983`
The data seems to be nowhere near. So which one of the two is correct?

**Comment by SabitovIlyas** [comment]
I have same trouble. I got bar from rest-api:
https://www.bitmex.com/api/v1/trade/bucketed?binSize=1h&partial=false&symbol=XBTUSD&count=1&reverse=false&startTime=2018-10-29+12:00:00

Bar is "timestamp":"2018-10-29T12:00:00.000Z, open":6400,"high":6400.5,"low":6225,"close":6341.5.
But on chart this bar has timestamp 2018-10-29T11:00:00.000Z.

Is it bug or feature?

**Comment by Wolfofcryptos** [comment]
same problem

**Comment by v0l0dia** [comment]
> Bar is "timestamp":"2018-10-29T12:00:00.000Z, open":6400,"high":6400.5,"low":6225,"close":6341.5.
> But on chart this bar has timestamp 2018-10-29T11:00:00.000Z.

It's not a bug - it's because bitmex Trades API returns timestamp that is equal to Period's closing time. It's stated in their API explorer:

> Implementation Notes
> Timestamps returned by our bucketed endpoints are the END of the period, indicating when the bucket was written to disk. Some other common systems use the timestamp as the beginning of the period. Please be aware of this when using this endpoint

**Comment by slot-29** [comment]
> Estoy tratando de obtener las últimas 20 velas de 5 minutos de Bitmex, pero cuando ingreso esta misma solicitud a través del Explorador de Bitmex ( https://www.bitmex.com/api/explorer/ ) o comparo los resultados con el gráfico, difieren . 
> ¿Hay alguna razón para esto?
> 
> ```
> .client.Trade.Trade_getBucketed(
>             binSize='5m',
>             partial=True,
>             symbol='XBT',
>             count=20,  
>             reverse=True
>         ).result()[0]
> ```

Hello, although this is old ... I need to solve it ... bitmex does not give me the last candle but the previous one ... I don't know why ... nor how to solve it ...

**Comment by jjalonso** [comment]
any update in knowledge?

**Comment by jeuxdemains** [comment]
Use the endTime parameter to access certain candle at specific time.
It's a bit confusing but the timestamp is the time at which the data is written down (which is when the candle is closing).

**Comment by novecento** [comment]
uhm, is this reliable? I mean, can I be sure in next "timeperiod" candle will be written?

**Comment by tianrenhefa** [comment]
has anyone fixed this problem? i've got problem in conecting to websocket and have to rely on rest api to get the last candle of 1m
